### PR TITLE
MWPW-126504 offer selector preview

### PR DESCRIPTION
**Pull request by VKniaz**
Added additional attributes to offer selector preview

Resolves: [MWPW-126504](https://jira.corp.adobe.com/browse/MWPW-126504)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/tools/commerce?martech=off
- After: https://MWPW-126504--milo--adobecom.hlx.page/tools/commerce?martech=off

### CODE REVIEW

**Comment by honstar** [comment]
@VKniaz : looks like there is a unit-test failure in `test/blocks/commerce/commerce.test.js`. Can you please check? Thx

File: libs/blocks/commerce/commerce.js [hunk 1]
```diff
@@ -56,16 +56,41 @@ export const filterOfferDetails = (o) => {
...
   }
   const offerDetailsList = document.createElement('ul');
   offerDetailsList.className = 'offer-details';
   const offer = filterOfferDetails(of);
+  const promotionCode = searchParams.get('promo');
+  const type = searchParams.get('type');
+  offer.type = type;
+  if (type === 'checkoutUrl') {
+    offer.cta = searchParams.get('text');
+  }
```
Comment (JasonHowellSlavin) [inline libs/blocks/commerce/commerce.js:89]: could consolidate: 

```suggestion
  offer.type = searchParams.get('type');
  if (offer.type === 'checkoutUrl') {
    offer.cta = searchParams.get('text');
  }
```

File: libs/blocks/commerce/commerce.js [hunk 2]
```diff
@@ -82,13 +107,21 @@ export const decorateOfferDetails = async (el, of, searchParams) => {
   const checkoutLink = document.createElement('a');
   checkoutLink.textContent = 'Checkout link';
   const perpetual = searchParams.get('perp') === 'true' || undefined;
-  const promotionCode = searchParams.get('promo') || undefined;
+
```
Comment (JasonHowellSlavin) [inline libs/blocks/commerce/commerce.js:110]: did you mean to add this new line?

**Review by JasonHowellSlavin — COMMENTED** [review]
Few nit picky things nothing major. 

Could probably remove the `feat()` from the PR title
feat(MWPW-126504):offer selector preview -> MWPW-126504 offer selector preview 

Do you need to modify any tests for additional functionality?

File: libs/blocks/commerce/commerce.js (hunk 2 above)
Comment (VKniaz) [reply libs/blocks/commerce/commerce.js]: removed line

File: libs/blocks/commerce/commerce.js [hunk 3]
```diff
@@ -82,13 +107,21 @@ export const decorateOfferDetails = async (el, of, searchParams) => {
...
   const options = omitNullValues({
     perpetual,
     promotionCode,
     wcsOsi: searchParams.get('osi'),
     ...getCheckoutContext(searchParams, getConfig()),
   });
+  const clearButton = buildClearButton();
```
[the remark this answers is not in the document]
Comment (VKniaz) [reply libs/blocks/commerce/commerce.js:117]: done

**Review by JasonHowellSlavin — APPROVED** [review]

**Comment by honstar** [comment]
@VKniaz : the codecov check currently fails, can you please add test coverage as per https://github.com/adobecom/milo/pull/902/checks?check_run_id=14617610719? Thanks

File: libs/blocks/commerce/commerce.js [hunk 4]
```diff
@@ -91,6 +114,14 @@ export const decorateOfferDetails = async (el, of, searchParams) => {
   });
   const checkoutUrl = buildCheckoutButton(checkoutLink, options);
   checkoutUrl.target = '_blank';
+  const clearButton = buildClearButton();
+
+  clearButton.addEventListener('click', () => {
```
Comment (mokimo) [inline libs/blocks/commerce/commerce.js:119]: If you already have `buildClearButton` you could have the event listener live in there too

**Review by mokimo — APPROVED** [review]

**Pull request merged**